### PR TITLE
Hardsuit Access and Emag

### PR DESCRIPTION
**Pull request by Geevies**
* You can now attack a hardsuit unit with any item to try and break open the access panel.
* Added some text prompts to hardsuit hacking.
* Hardsuits now spark when you electrify them.

Also cleans up a lot of RIG code in the second commit.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by Arrow768** [comment]
> Emagging a hardsuit unit now lets you add any hardsuit item to it, regardless of class restriction.

Not in favor of that, as these restrictions have been added to prevent a specific antag gameplay style which was rather annoying to deal with.

File: code/modules/clothing/spacesuits/rig/rig_attackby.dm [hunk 1]
```diff
@@ -153,53 +153,63 @@
...
 
 	// If we've gotten this far, all we have left to do before we pass off to root procs
 	// is check if any of the loaded modules want to use the item we've been given.
 	for(var/obj/item/rig_module/module in installed_modules)
-		if(module.accepts_item(W,user)) //Item is handled in this proc
+		if(module.accepts_item(W, user)) //Item is handled in this proc
 			return
-	..()
 
+	if(!open)
```
Comment (NonQueueingMatt) [inline code/modules/clothing/spacesuits/rig/rig_attackby.dm:197]: this shouldn't work if it's on your bag

Comment (NonQueueingMatt) [reply code/modules/clothing/spacesuits/rig/rig_attackby.dm:197]: this is also a bit too risk-free and too fast for my tastes, should be a do_after with a chance of things going wrong tbh

**Comment by NonQueueingMatt** [comment]
Unaddressed changes and conflicts.

**Pull request closed without merging**